src/shims/foreign_items.rs

```diff
@@ -160,13 +163,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             // Normally, this will be either `libpanic_unwind` or `libpanic_abort`, but it could
             // also be a custom user-provided implementation via `#![feature(panic_runtime)]`
             "__rust_start_panic" | "__rust_panic_cleanup"=> {
-                // FIXME we might want to cache this... but it's not really performance-critical.
-                let panic_runtime = tcx
-                    .crates()
-                    .iter()
-                    .find(|cnum| tcx.is_panic_runtime(**cnum))
-                    .expect("No panic runtime found!");
-                let panic_runtime = tcx.crate_name(*panic_runtime);
+                // This replicates some of the logic in `inject_panic_runtime`.
+                // FIXME: is there a way to reuse that logic?
```

Comment:
we could have a `start_panic_instance` function on `Session` in rustc

Reply:
Or maybe we could somehow figure out "which crate that was added as a dependency is the panic runtime". I though that's what the previous code was doing, but it still saw the panic_unwind crate even with `-C panic=abort` so clearly that was not the right approach.